observability-kit-starter/src/test/java/com/vaadin/observability/ObservabilityHandlerTest.java

```diff
@@ -0,0 +1,201 @@
+package com.vaadin.observability;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.server.VaadinRequest;
+import com.vaadin.flow.server.VaadinResponse;
+import com.vaadin.flow.server.VaadinSession;
+import com.vaadin.flow.shared.ApplicationConstants;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class ObservabilityHandlerTest {
+    @Test
+    public void ensureInstalled_handlerIsInstalled() {
+        UI ui = mock(UI.class);
+        VaadinSession session = mock(VaadinSession.class);
+        when(ui.getSession()).thenReturn(session);
+
+        ObservabilityHandler.ensureInstalled(ui);
+
+        verify(session).addRequestHandler(any());
```

Comment:
nit: I would check that the added request handler is of the expected type 
```suggestion
        verify(session).addRequestHandler(any(ObservabilityHandler.class));
```